src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -291,6 +310,30 @@ class LinalgBackendEigen : public LinalgBackendBase
 		return (typename SGVector<T>::EigenVectorXtMap(a)).dot(typename SGVector<T>::EigenVectorXtMap(b));
 	}
 
+	/** Eigen3 matrix elementwise product method */
+	template <typename T>
+	SGMatrix<T> element_prod_impl(const SGMatrix<T>& a, const SGMatrix<T>& b) const
+	{
+		SGMatrix<T> c(a.num_rows, a.num_cols);
+		typename SGMatrix<T>::EigenMatrixXtMap a_eig = a;
+		typename SGMatrix<T>::EigenMatrixXtMap b_eig = b;
+		typename SGMatrix<T>::EigenMatrixXtMap c_eig = c;
+
+		c_eig = a_eig.array() * b_eig.array();
+		return c;
+	}
+
+	/** Eigen3 matrix in-place elementwise product method */
+	template <typename T>
+	void element_prod_impl(SGMatrix<T>& a, SGMatrix<T>& b, SGMatrix<T>& result) const
+	{
+		typename SGMatrix<T>::EigenMatrixXtMap a_eig = a;
+		typename SGMatrix<T>::EigenMatrixXtMap b_eig = b;
+		typename SGMatrix<T>::EigenMatrixXtMap result_eig = result;
+
+		result_eig = a_eig.array() * b_eig.array();
```

Comment:
It is dangerous whenever you are overwriting memory that is used for the computation.
This is why eigen3 often allocates temp memory. Make sure you really understand the mechanics here. Array is just a view onto matrices that allows for element wise operations (different operator * overloading for example)